apiserver/application/application.go

```diff
@@ -1048,7 +1044,31 @@ func (api *API) sameControllerOfferedApplication(url *jujucrossmodel.Application
 		return fail(errors.Errorf("unexpected: %d matching offers for %q", len(offers), url.ApplicationName))
 	}
 
+	// Check the permissions - a user can access the offer if they are an admin
+	// or they have consume access to the offer.
+	isAdmin := false
```

Comment:
Or maybe the whole section into a `HasOfferAccess(st, offerName, apiUser)` method?